### ISSUE TITLE
Failed building VG

### ISSUE THREAD

**Issue by RenzoTale88** [vgteam/vg#2491]
Dear vg team,
since I need to build hal2vg, I've been trying to build vg from sources.
I've been following the guide provided, installing all the dependencies on ubuntu 16.04.
I've managed to install it, but with several issues that I'd like to send you, maybe they can be useful to you or to who is having my same issues.

I've done as follow:

```
git clone --recursive https://github.com/vgteam/vg.git
cd vg
# Install all but protobuf 3
sudo apt-get install build-essential git cmake pkg-config libncurses-dev libbz2-dev  \
                     libjansson-dev automake libtool jq bc rs curl unzip redland-utils \
                     librdf-dev bison flex gawk lzma-dev liblzma-dev liblz4-dev \
                     libffi-dev libcairo-dev

# Install Protobuf 3
wget https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protobuf-all-3.6.1.tar.gz
tar xvfz protoc-3.6.1-linux-x86_64.zip && cd protobuf-3.6.1
./configure
make 
make check
sudo make install
sudo ldconfig

# Compile VG
. ./source_me.sh && make

```
 Despite that, the whole compilation would fail with the default gcc compiler (GCC 5.4.0). 
To fix it, I had to install gcc v7.4.0 and CMake >3.9 and repeat the compilation. Still, I was getting an error when trying to import BooPHF.h. To fix that, I had to:
```
cp vg/deps/BBHash/BooPHF.h vg/include
```
With these changes, I was able to compile vg without errors.
Attached, you can find the error I got when compiling with gcc 5.4.0.
Hope this is of any help,
All the best,

Andrea

[VGerror.txt](https://github.com/vgteam/vg/files/3667437/VGerror.txt)

**Comment by jeizenga** [comment]
Huh. @adamnovak, any idea why this wouldn't copy `BooPHF.h`?
https://github.com/vgteam/vg/blob/master/Makefile#L425

**Comment by adamnovak** [comment]
It looks like it should.

Do you maybe have a case problem?

Or are you missing a dependency between whatever actually uses the
header and that Makefile rule?

I'm not sure why it doesn't work with GCC 5.4 either, other than that
we haven't been testing with GCC 5.4 in a while.

**Comment by RenzoTale88** [comment]
I got no clue. I've tried to compile on a separate ubuntu 16.04 machine, with GCC9 installed instead of GCC7. It compile the library just fine, but then again it do not copy it. After copying it to the include folder, everything run just fine.

**Comment by jeizenga** [comment]
Yeah, and at least by eye the dependencies should go
```
all: $(BIN_DIR)/$(EXE)
$(BIN_DIR)/$(EXE): ... $(DEPS) ... 
DEPS += $(INC_DIR)/BooPHF.h
$(INC_DIR)/BooPHF.h: $(BBHASH_DIR)/BooPHF.h
```

**Comment by LilithElina** [comment]
How did you do that? When I try to `make` vg, I end up with 

> make: *** No rule to make target 'deps/sdsl-lite/lib/*.cpp', needed by 'lib/libsdsl.a'.  Stop.

**Comment by ekg** [comment]
Did you clone with `git clone --recursive` ?

**Comment by LilithElina** [comment]
Yes, and I thought except for the time out of elfutils everything was cloned, but all the directories in deps/ are empty...

**Comment by LilithElina** [comment]
Forget the question, I know what you did: you did not face company network restrictions. Thanks to randomly clicking Google results I found the solution [here](https://github.com/angular/angular-phonecat/issues/141): `git config --global url."https://".insteadOf git://` helped me change the elfutils link and then it worked fine.

**Comment by kojix2** [comment]
@LilithElina The above command saved the day. Perhaps the problem is related to the proxy. Thank you.

**Comment by faithokamoto** [comment]
Seems this was handled.
